### DISCUSSION TITLE
how to get tags in export template jinja2

### DISCUSSION THREAD

**Discussion by henal3112** [netbox-community/netbox#15074] [Q&A] [answered]
Hi,

I want to create export template 

NAME,SITE,Tag
{% for device in queryset %}
{{device.name}},{{device.site}},{{device.tags}}
{% endfor %}

but I cannot get the tags part as I have created following tag , so How can I filter EOL tag devices 
API shows as below..
 "tags": [
                {
                    "id": 1,
                    "url": "http://0.0.0.0:8000/api/extras/tags/1/",
                    "display": "EOL",
                    "name": "EOL",
                    "slug": "eol",
                    "color": "795548"
                }

**Comment by candlerb** [comment, marked as answer]
IIRC, the `tags` attribute has a method `names()` to give just the tag names, i.e. `{{ device.tags.names() }}`. Or `slugs()` if you want the tag slugs instead of tag names.

The way to explore these things is with [nbshell](https://docs.netbox.dev/en/stable/administration/netbox-shell/).

```
# /opt/netbox/venv/bin/python /opt/netbox/netbox/manage.py nbshell
...
>>> o = IPAddress.objects.get(id=302)
>>> o.tags
<taggit.managers._TaggableManager object at 0x7f3a44fdc160>
>>> o.tags.all()
<RestrictedQuerySet [<Tag: proxy>]>
>>> list(o.tags.all())
[<Tag: proxy>]
>>> o.tags.names()
<RestrictedQuerySet ['proxy']>
>>> list(o.tags.names())
['proxy']
>>> o.tags.slugs()
<RestrictedQuerySet ['proxy']>
>>> list(o.tags.slugs())
['proxy']
```

**Comment by henal3112** [comment]
Thanks @candlerb ,

I am running netbox on docker instance so how can I access nbshell on docker ?

**Reply by KGT1** [reply]
If you use docker compose:
something like this:
`docker compose exec -it --user unit netbox ./manage.py nbshell`

**Comment by henal3112** [comment]
I am getting following ouput if i use tags.names


Server01,Germany,active,prodgermany01,.<RestrictedQuerySet ['eol']>

how can I remove .<RestrictedQuerySet ?

**Reply by candlerb** [reply]
Try `{{ device.tags.names() | list }}` or maybe `{{ device.tags.names() | join("+") }}`

nbshell on docker: I guess you'd start with something like `docker exec -it <container> -- /bin/sh` (untested) to get a shell inside the container.
